types/address.go

```diff
@@ -256,6 +264,31 @@ func (aa AccAddress) Format(s fmt.State, verb rune) {
 	}
 }
 
+// LengthPrefixedAddress prefixes the address bytes with its length, this is used
+// for variable-length components in store keys. Note: All addresses should be
+// max 255 bytes, or else this function panics.
+func LengthPrefixedAddress(bz []byte) ([]byte, error) {
+	if len(bz) == 0 {
+		return bz, nil
+	}
+
+	if len(bz) > maxAddrLen {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "address length should be max %d bytes, got %d", maxAddrLen, len(bz))
+	}
+
+	return append([]byte{byte(len(bz))}, bz...), nil
+}
```

Comment:
In https://github.com/cosmos/cosmos-sdk/pull/8415 I've created an `address` package. I would prefer to put this functions in that package and remove the `Address` suffix (so it will be named `address.LengthPrefixed ` instead of `types.LengthPrefixedAddress` (or `address.MkLengthPrefixed`). To options to move forward:
1. merge https://github.com/cosmos/cosmos-sdk/pull/8415 and rebase
2. create the package and move these functions there in this PR
3. do it in other PR.

My preference is 1 or 2.

Reply:
I think `address.LengthPrefixedStoreKey` works